core/drivers/nvmem/nvmem_huk.c

```diff
@@ -19,19 +19,20 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 {
 	TEE_Result res = TEE_ERROR_GENERIC;
 	uint8_t *huk = NULL;
-	size_t len = 0;
 
-	res = nvmem_cell_malloc_and_read(huk_cell, &huk);
-	if (res)
-		goto out_free_cell;
-
-	if (len != HW_UNIQUE_KEY_LENGTH) {
+	if (huk_cell->len < HW_UNIQUE_KEY_LENGTH) {
 		res = TEE_ERROR_GENERIC;
 		goto out_free_cell;
 	}
 
+	res = nvmem_cell_malloc_and_read(huk_cell, &huk);
+	if (res)
+		goto out_free_cell;
+
 	memcpy(hwkey->data, huk, HW_UNIQUE_KEY_LENGTH);
```

Comment:
As the data copied is counted with the unit byte, the `bits` need to be replaced with `bytes` in the commit messages of 31059bc & 086e75a

Reply:
Hello @TonyHan11,

> As the data copied is counted with the unit byte, the `bits` need to be replaced with `bytes` in the commit messages of [31059bc](https://github.com/OP-TEE/optee_os/commit/31059bc92c18a8ca0e73cac4ac139faa0bb68c9e) & [086e75a](https://github.com/OP-TEE/optee_os/commit/086e75a460946d0577d7e93fbc5cfbdc284e4163)

Thank you, I corrected commit messages.

Kind regards, 
Thomas